### PR TITLE
Add models for new TextInput question type

### PR DESCRIPTION
**Pull request by makqien**
Creation of a new TextInput question type to support GCE A-Level General Paper Comprehension questions.

Possibly to migrate TextResponse questions and answers over to TextInput once TextInput is complete.

### CODE REVIEW

**Comment by weiqingtoh** [comment]
> Possibly to migrate TextResponse questions and answers over to TextInput once TextInput is complete.

Actually, why not just modify the current set of tables and models?

**Comment by makqien** [comment]
>Actually, why not just modify the current set of tables and models?

I'm afraid I may break something in the existing TextResponse questions and answers at this early stage.

File: app/models/course/assessment/question/text_input_solution.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  end
+
+  def initialize_duplicate(duplicator, other)
+    self.point = duplicator.duplicate(other.point)
+  end
+
+  private
+
+  def strip_whitespace_solution
+    solution.each { |s| s.strip! } unless solution.empty?
```
[the remark this answers is not in the document]
Comment (fonglh) [reply app/models/course/assessment/question/text_input_solution.rb]: `solution.each(&:strip!)` should work

**Comment by allenwq** [comment]
> I'm afraid I may break something in the existing TextResponse questions and answers at this early stage.

May I suggest you PR this into a feature branch? finally we can merge the feature and mater when it's complete. So you can directly work on text response questions, currently there are quite a bit of things are copied from text response, which is unnecessary for double reviewing.

**Comment by makqien** [comment]
>May I suggest you PR this into a feature branch? finally we can merge the feature and mater when it's complete. So you can directly work on text response questions, currently there are quite a bit of things are copied from text response, which is unnecessary for double reviewing.

@allenwq Okay. After #2629 is merged, will need your help to create a new feature branch and I'll PR from there. Will close this PR for now.

**Comment by allenwq** [comment]
@makqien thanks! 👍

**Pull request closed without merging**